cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -270,11 +270,11 @@ private void tagsQuerying(HttpRequest request, HttpResponder responder, List<Str
   }
 
   private void contextQuerying(HttpRequest request, HttpResponder responder,
-                               String context, String metric, String groupBy) {
+                               String context, List<String> metrics, String groupBy) {
     try {
       List<String> groupByTags = parseGroupBy(groupBy);
       MetricQueryResult queryResult = executeQuery(request, parseTagValuesAsMap(context),
-                                                   groupByTags, ImmutableList.of(metric));
+                                                   groupByTags, metrics);
```

Comment:
its an empty list and in executeQuery() we check if the list is empty for metrics and throw IllegalArgumentException there which gets caught and thrown as BadRequest